plugin.php

```diff
@@ -552,6 +587,21 @@ function rest_sanitize_request_arg( $value, $request, $param ) {
 			return (int) $value;
 		}
 
+		if ( 'array' === $args['type'] ) {
+			// A comma-separated array of IDs could be acceptable instead of an array.
+			if ( is_string( $value ) && isset( $args['allow_csv'] ) && $args['allow_csv'] ) {
+				$value = wp_parse_id_list( $value );
+			}
+
+			$value = (array) $value;
+
+			// Map values to the correct type, defaulting to string as the type.
+			$type = isset( $args['items']['type'] ) ? $args['items']['type'] : 'string';
+			foreach ( $value as &$_value ) {
+				settype( $_value, $type );
```

Comment:
Rather than using `settype` we should use our own validation systems because `settype` can lead to the wrong information.  We can get away with `settype` though as most of the endpoints only deal with arrays of strings or integers.